File: 프로그래머스/2/12924. 숫자의 표현/숫자의 표현.py
'''
n을 연속된 자연수로 표현하는 법

'''
def solution(n):
    if n == 1: return 1
    left, right = 1, 2
    summ = left+right
    cnt = 0
    while left <= right:
        if summ <= n:
            if summ == n: 
                cnt+=1
            right+=1
            summ+=right
        elif summ > n: # n보다 크다면 left 줄이기
            summ -= left
            left+=1
    
    return cnt

File: 프로그래머스/2/12924. 숫자의 표현/숫자의 표현.java
class Solution {
    public int solution(int n) {
    
        int answer = 0;
        if (n == 1){
            return 1;
        }

        int left = 1;
        int right = 2;
        int summ = 3;
        while (left < right && right < n) {
            if (summ == n){
                answer+=1;
                summ -= left;
                left += 1;
            }
            else if (summ < n){
                right += 1;
                summ += right;
            }
            else if (summ > n){
                summ -= left;
                left += 1;
            }
        }

        return answer+1;
    }
    
}